### DISCUSSION TITLE
Can I use this library only to plot my truss structures?

### DISCUSSION THREAD

**Discussion by Martin15135215** [JWock82/Pynite#180] [Q&A] [answered]
I want to plot my simple truss (2D) and space truss (3D). Can I use this library only for this use case?

**Comment by JWock82** [comment, marked as answer]
Yes: you can render a model with or without its loads without having to analyze it.

Thanks,

Craig Brinck
________________________________
From: Martin15135215 ***@***.***>
Sent: Tuesday, December 26, 2023 5:00 AM
To: JWock82/PyNite ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [JWock82/PyNite] Can I use this library only to plot my truss structures? (Discussion #180)


I want to plot my simple truss (2D) and space truss (3D). Can I use this library only for this use case?

**Comment by Martin15135215** [comment]
Thanks, I saw no examples in the documentation. However, I found it on GitHub, in the folder examples, the function

https://github.com/JWock82/PyNite/blob/90d6fdc9fab57cc46e5d088fdf620988c901b50e/Examples/Space%20Truss%20-%20Nodal%20Load.py#L70-L73

Sadly, it opens a new window instead of showing in my Jupyter notebook.

I tried than the function ViewInteractiveWidget from here https://github.com/Kitware/ipyvtklink, and changed the function Visualization.render_model to return the window value and deleted the line with window.Render()
 etc., so I could, it input it into the function ViewInteractiveWidget. However, there was an error message.

My goal was when I save the ipynb as HTML or PDF etc, there is still the 3D plot in it.

**Reply by JWock82** [reply]
It can plot to Jupyter Notebooks. The `Visualization.render_model` method is an old way of doing things that it's about time to deprecate. Pynite has a `Renderer` object that offers a lot of options for rendering. I use Pynite with Jupyter all the time. Here's an example:

    # Import the `Renderer` class
    from PyNite.Visulization import Renderer

    # Create a new renderer for the model
    my_renderer = Renderer(my_model)

    # Adjust any renderer settings you want to
    my_renderer.window_width = 750
    my_renderer.window_height = 500
    my_renderer.annotation_size = 5

    # Have the renderer produce a screenshot for Jupyter Notebook
    my_renderer.screenshot()

**Reply by JWock82** [reply]
With the code above, The renderer will let you position the screenshot. Once you've done that and close out, it will print to Jupyter Notebook.

**Reply by Martin15135215** [reply]
I tried it with the above example

![image](https://github.com/JWock82/PyNite/assets/110968398/60f06082-2a57-4ea5-8009-9da1db9fabee)

**Reply by JWock82** [reply]
from PyNite.Visualization import Renderer

**Reply by Martin15135215** [reply]
The below image taken with the old function

![image](https://github.com/JWock82/PyNite/assets/110968398/c4379a79-84e0-49cd-9d45-b46ac7cf73a3)

the below picture with the new class

![image](https://github.com/JWock82/PyNite/assets/110968398/79d53838-2328-40b2-823c-9bec68a900ae)

**Reply by JWock82** [reply]
Change the annotation scale

**Reply by JWock82** [reply]
For example:  
  
    my_renderer.annotation_size = 0.1

**Reply by JWock82** [reply]
Or to match your prior rendering: 0.05

**Reply by JWock82** [reply]
Any luck?

**Reply by Martin15135215** [reply]
Thanks, that did the trick:

![image](https://github.com/JWock82/PyNite/assets/110968398/d8c116f1-adb7-4228-a5a7-72c575e1a14f)

However, my image is now black:

![image](https://github.com/JWock82/PyNite/assets/110968398/d1b67d33-923b-4135-84af-0843493b2b65)

**Reply by JWock82** [reply]
That doesn't seem right. What operating system are you on? Unix?

**Reply by Martin15135215** [reply]
Yes, Debian 12

**Reply by JWock82** [reply]
Hmmm... I know there was an issue someone reported between VTK and Unix. They found a solution. I'm guessing this has something to do with that. I've never had trouble with this on Windows.

**Reply by JWock82** [reply]
Have a look at issue #139 and see if it helps.

**Reply by Martin15135215** [reply]
I appreciate it, I will look into it later.
